src/Compilers/Core/Portable/SourceGeneration/GeneratorDriver.cs

```diff
@@ -52,7 +52,12 @@ public GeneratorDriver RunGenerators(Compilation compilation, CancellationToken
             return FromState(state);
         }
 
+        // https://github.com/dotnet/roslyn/issues/46623
+        [Obsolete("Use RunGeneratorsAndUpdateCompilation")]
         public GeneratorDriver RunFullGeneration(Compilation compilation, out Compilation outputCompilation, out ImmutableArray<Diagnostic> diagnostics, CancellationToken cancellationToken = default)
+            => RunGeneratorsAndUpdateCompilation(compilation, out outputCompilation, out diagnostics, cancellationToken);
```

Comment:
Why forward this one, but not forward the renamed contexts? It seems that we should either forward everything or nothing, otherwise things will be in a half-working state.

Reply:
will make it throw instead.